cloudinit/sources/DataSourceVMware.py

```diff
@@ -262,6 +354,45 @@ def network_config(self):
             }
         return self.metadata["network"]["config"]
 
+    def advertise_update_events(self, cfg):
+        default_events: Dict[EventScope, Set[EventType]] = copy.deepcopy(
+            self.default_update_events
+        )
+        user_events: Dict[EventScope, Set[EventType]] = userdata_to_events(
+            cfg.get("updates", {})
+        )
+        enabled_events = util.mergemanydict(
+            [
+                copy.deepcopy(user_events),
+                copy.deepcopy(default_events),
+            ]
+        )
+        return advertise_update_events(
+            self.supported_update_events,
+            enabled_events,
+            self.rpctool,
+            self.rpctool_fn,
+        )
+
+    def get_network_drivers(self):
+        network_drivers = DEFAULT_NETWORK_DRIVERS
+
+        if self.metadata and "network-drivers" in self.metadata:
+            network_drivers = self.metadata["network-drivers"]
+
+        if isinstance(network_drivers, bytes):
```

Comment:
Out of curiosity, are the supported formats to specify the network drivers documented somewhere?  For example, you are assuming that the strings delimited by `|` are white-space trimmed.  I am wondering if that's a same assumption.

Reply:
Good question, and good point. I can trim the tokens of WS. As for the format, there's a whole section that has been added to the documentation at https://github.com/canonical/cloud-init/pull/6063/files#diff-667e7d12ae3ab4bebf1a4cb838bd46b2557da884106043b5456f62794f7352d9R372-R397. It describes a list. I added the ability to parse a `|`-delimited string as well just in case people were to copy/paste something directly from the udev rules, ex. https://github.com/canonical/cloud-init/blob/3da7eca873a7c26aec111d4af8d8364a71437b48/cloudinit/sources/DataSourceEc2.py#L58-L63.